pkg/hub/submarineragent/controller.go

```diff
@@ -692,9 +692,33 @@ func (c *submarinerAgentController) createGNConfigMapIfNecessary(ctx context.Con
 		return errors.Wrapf(err, "error creating globalnet configmap on Broker")
 	}
 
+	// Ideally we should add option to have custom labels when creating configmap
+	if err := c.updateGnConfigMapWithBackupLabel(ctx, brokerNamespace); err != nil {
+		return errors.Wrapf(err, "error adding backup label to globanet configmap on Broker")
+	}
+
 	return nil
 }
 
+func (c *submarinerAgentController) updateGnConfigMapWithBackupLabel(ctx context.Context, namespace string) error {
+	gnConfigMap, err := globalnet.GetConfigMap(ctx, c.controllerClient, namespace)
+
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+
+	if gnConfigMap != nil {
+		gnConfigMap.ObjectMeta.Labels[BackupLabel] = "submariner"
+		err = c.controllerClient.Update(ctx, gnConfigMap)
+	}
+
+	return err
```

Comment:
Shouldn’t this be in a retry loop?

Reply:
We are the only writer of this resource and caller will end up retrying so didn't do that. Still, good idea to retry here than repeat the entire sync loop.

Reply:
If there is an error while adding the label, the sync logic will not help. When the code finds that the configMap already exists, it will return in Line 666